.github/workflows/test.yml

```diff
@@ -0,0 +1,64 @@
+# things not included
+# git depth (not an issue with gh actions)
+# languaage
+# notifications - no email notifications set up
+# completely lost on stages name/if section
+
+
+name: pytest
+on:
+  pull_request:
+    branches:
+    - '*'
+jobs:
+  default:
+    name: Pytest on ${{ matrix.python-version }}, ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest']
+        python-version: [3.7]
+    timeout-minutes: 30
+    env:
+      DESC: "Python 3.6 tests"  # f strings? is this even necessary?
+      HV_REQUIREMENTS: "unit_tests"
+      PYTHON_VERSION: ${{ matrix.python-version }}
+      PKG_TEST_PYTHON: "--test-python=py37 --test-python=py27"
+      CHANS_DEV: "-c pyviz/label/dev -c bokeh"
+      CHANS: "-c pyviz"
+      MPLBACKEND: "Agg"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: pre_install
+        run: |
+          python -m pip install --upgrade pip
+          pip install pyctdev flake8 nose
+          doit miniconda_install
+      - name: conda setup
+        run: |
+          export PATH="$HOME/miniconda/bin:$PATH" && hash -r
```

Comment:
In order for this to transfer to the next steps (test_examples, coverall) you should add a 
```
##vso[task.prependpath]($HOME)/miniconda3/bin
```

like in this [stackoverflow answer](https://stackoverflow.com/questions/57322531/how-to-set-path-in-azure-pipeline-using-prependpath)

You might be interested in using [this action](https://github.com/marketplace/actions/setup-conda) which is supposed to set this all up for you, and works across different OSes

Reply:
Thanks @mattip!

Reply:
It turns out there are a few options on github actions for conda, [here](https://github.com/marketplace/actions/setup-miniconda) is another one that uses mini-conda instead of conda

Reply:
which is mentioned in the comments above too.